client_plugin/drivers/shared/shared_driver.go

```diff
@@ -124,11 +124,12 @@ func NewVolumeDriver(cfg config.Config, mountDir string) *VolumeDriver {
 	}
 
 	// initialize built-in etcd cluster
-	d.kvStore = etcdops.NewKvStore(d.dockerOps)
-	if d.kvStore == nil {
+	etcdKVS := etcdops.NewKvStore(d.dockerOps)
```

Comment:
I also had the same understanding...
But with the original code, d.kvStore won't be nil even if etcdops.NewKvStore returns nil. I think the reason is because d.kvStore is an interface instead of a struct.
The fix is exactly to handle this issue.